federation/pkg/federation-controller/util/federated_informer.go

```
@@ -0,0 +1,408 @@
+	federation_release_1_4 "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	api "k8s.io/kubernetes/pkg/api"
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	pkg_runtime "k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
+)
+
+const (
+	clusterSyncPeriod = 10 * time.Minute
+	UserAgentName     = "federation-service-controller"
+)
+
+// FederatedReadOnlyStore is an overlay over multiple stores created in federated clusters.
+type FederatedReadOnlyStore interface {
+
+	// Returns all items in the store.
+	List() []interface{}
+
+	// GetByKey returns the item stored under the given key in the specified cluster (if exist).
+	GetByKey(clusterName string, key string) (item interface{}, exists bool)
+
+	// Returns the items stored under the given key in all clusters.
+	GetFromAllClusters(key string) []interface{}
+
+	// Checks whether stores for all clusters form the lists (and only these) are there and
+	// are synced. This is only a basic check whether the data inside of the store is usable.
+	// It is not a full synchronization/locking mechanism it only tries to ensure that out-of-sync
+	// issues occur less often.	All users of the interface should assume
+	// that there may be significant delays in content updates of all kinds and write their
+	// code that it doesn't break if something is slightly out-of-sync.
+	Synced(clusters []string) bool
+}
+
+// An interface to access federation members and clients.
+type FederationView interface {
+	// GetClientsetForCluster returns a clientset for the cluster, if present.
+	GetClientsetForCluster(clusterName string) (federation_release_1_4.Interface, error)
+
+	// GetClusers returns all clusters for which the sub-informers are run.
+	GetClusters() []*federation_api.Cluster
```

Looking into implementation below, I think we shouldn't hide errors. So this function should also return error in my opinion.